Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1

```diff
@@ -289,7 +295,14 @@ function GetWorkflowContentWithChangesFromSettings {
         ModifyPullRequestHandlerWorkflow -yaml $yaml -repoSettings $repoSettings
     }
 
-    if ($baseName -ne "UpdateGitHubGoSystemFiles" -and $baseName -ne 'Troubleshooting') {
+    # Workflows in the blacklist may only run on runners listed in the whitelist
+    if($workflowBlacklist -contains $baseName) {
+        if($runnerWhitelist -notcontains $repoSettings."runs-on") {
+            $modifyRunsOnAndShell = $false
+        }
+    }

```

Comment:
I am not sure that the end 2 end tests covers this scenario. If Update AL-Go system files is set to Ubuntu-latest and it for some reason doesn't work - people will be dead in the water.
I will try to run end 2 end tests on this PR to see whether it is covered.

Reply:
This wouldn't allow people to use self-hosted runners for AL-Go updates and Troubleshooting

Reply:
That is already the case today. These workflows can only run on GitHub hosted agents to avoid deadlocks if self hosted agents are not working (where they need to run update al-go system files to resolve)

Reply:
I was right in assuming that no end 2 end test will cover the scenario that Update AL-Go System Files actually works on ubuntu-latest. I will create a PR (on your PR) with an end 2 end test

Reply:
Have modified end 2 end tests to use ubuntu for these workflows as well - and run all scenario tests on windows and linux - running end 2 end now - if they pass, I am OK with this.

Reply:
I think I have been running too many tests and exhausted my limits - will re-run later

Reply:
The Update AL-Go System Files actually didn't work properly under Linux due to this Powershell "bug": https://github.com/PowerShell/PowerShell/issues/6473#issuecomment-375930843

Under Linux, folders starting with a period are considered hidden and Resolve-Path cannot resolve a hidden path.
I have added a workaround to the PR and re-running end 2 end tests now.

Reply:
The consequence of this bug was that scripts and files directly under .github or under .AL-Go would not have been updated by Update AL-Go System Files.
Running a full end 2 end now https://github.com/microsoft/AL-Go/actions/runs/10088558197